src/Psalm/Internal/Analyzer/Statements/Expression/ClassConstAnalyzer.php

```diff
@@ -721,19 +721,27 @@ public static function analyzeAssignment(
 
             // Check assigned type matches docblock type
             if ($assigned_type = $statements_analyzer->node_data->getType($const->value)) {
-                if ($const_storage->type !== null
+                $const_storage_type = $const_storage->type;
+
+                if ($const_storage_type !== null
                     && $const_storage->stmt_location !== null
-                    && $assigned_type !== $const_storage->type
+                    && $assigned_type !== $const_storage_type
+                    // Check if this type was defined via a dockblock or type hint otherwise the inferred type
+                    // should always match the assigned type and we don't even need to do additional checks
+                    // There is an issue with constants over a certain length where additional values
+                    // are added to fallback_params in the assigned_type but not in const_storage_type
+                    // which causes a false flag for this error to appear. Usually happens with arrays
+                    && ($const_storage_type->from_docblock || $const_storage_type->from_property)
```

Comment:
I think this can be reduced down to just `$const_storage_type->from_docblock` but this will later cause issues with PHP 8.3 and constant type hinting. There needs to be a way to check if a type came from the docblock or from type hinting to keep this logic into the future.